backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/work/impl/SearchWork.java

```diff
@@ -56,8 +56,9 @@ protected R generateResult(ElasticsearchWorkExecutionContext context, Elasticsea
 	public static class Builder<R>
 			extends AbstractBuilder<Builder<R>> {
 
-		public static <T> Builder<T> create(JsonObject payload, ElasticsearchSearchResultExtractor<T> resultExtractor) {
-			return new Builder<>( payload, resultExtractor, true, false );
+		public static <T> Builder<T> create(JsonObject payload, ElasticsearchSearchResultExtractor<T> resultExtractor,
+				boolean ignoreShardFailures) {
```

Comment:
IMO either you consider this will one day be customized per-query and it should be passed through a setter in this builder (like `disableTrackTotalHits()`), or you consider it's just a global thing and it would probably make more sense to pass the information through the constructor of `ElasticsearchWorkFactory`.

Reply:
ohh and we probably want to flip the default in 6.2 to ignore these partial failures.. so then 7.0 needs an entry in the migration guide, right ?